test/oz-account-test.ts

```diff
@@ -0,0 +1,171 @@
+    console.log("Private and public key:", privateKey, publicKey);
+  });
+
+  it("should load an already deployed account with the correct private key", async function() {
+
+    const loadedAccount = await starknet.getAccountFromAddress(accountAddress, privateKey, "OpenZeppelin");
+
+    expect(loadedAccount.starknetContract.address).to.deep.equal(accountAddress);
+    expect(loadedAccount.privateKey).to.deep.equal(privateKey);
+    expect(loadedAccount.publicKey).to.deep.equal(publicKey);
+  });
+
+  it("should fail when loading an already deployed account with a wrong private key", async function() {
+    try{
+      await starknet.getAccountFromAddress(accountAddress, "0x0123", "OpenZeppelin");
+      expect.fail("Should have failed on passing an incorrect private key.");
+    } catch(err: any) {
+      expect(err.message).to.equal("The provided private key is not compatible with the public key stored in the contract.");
+    }
+  });
+
+  it("should invoke a function on another contract", async function() {
+    const { res: currBalance } = await account.call(mainContract, "get_balance");
+    const amount1 = 10n;
+    const amount2 = 20n;
+    await account.invoke(mainContract, "increase_balance", { amount1, amount2 });
+
+    const { res: newBalance } = await account.call(mainContract, "get_balance");
+    expect(newBalance).to.deep.equal(currBalance + amount1 + amount2);
+  });
+
+  it("should work with arrays through an account", async function() {
+    const { res } = await account.call(mainContract, "sum_array", { a: [1, 2, 3] });
+    expect(res).to.equal(6n);
+  });
+
+  it("should work with BigNumbers and tuples through an account", async function() {
+    // passing Points (1, 2) and (3, 4) in a tuple
+    const { res: sum } = await account.call(mainContract, "sum_points_to_tuple", {
+      points: [
+        { x: BigNumber.from(1), y: BigNumber.from(2) },
+        { x: 3, y: 4 }
+      ]
+    });
+    expect(sum).to.deep.equal([4n, 6n]);
+  });
+
+  // Multicall / Multiinvoke testing
+  it("should handle multiple invokes through an account", async function() {
+    const { res: currBalance } = await account.call(mainContract, "get_balance");
+    const amount1 = 10n;
+    const amount2 = 20n;
+
+    const invokeArray = [
+      {
+        toContract: mainContract,
+        functionName: "increase_balance",
+        calldata: {amount1, amount2}
+      },
+      {
+        toContract: mainContract,
+        functionName: "increase_balance",
+        calldata: {amount1, amount2}
+      },
+      {
+        toContract: mainContract,
+        functionName: "increase_balance",
+        calldata: {amount1, amount2}
+      }
+    ];
+
+    const txHashArray = await account.multiInvoke(invokeArray);
+    console.log(txHashArray);
+    const { res: newBalance } = await account.call(mainContract, "get_balance");
+    expect(newBalance).to.deep.equal(currBalance + 3n * (amount1 + amount2));
```

Comment:
I believe as little arithmetic should be used in assertions. So, `90n` may be used instead of `3n * (amount1 + amount2)`. Also, if the same can be applied for the other account and other occurrences..